napari/_qt/widgets/_tests/test_qt_play.py

```diff
@@ -155,3 +156,31 @@ def increment(e):
         view.dims.play(2, 20)
     view.dims.dims.events.current_step.disconnect(increment)
     assert not view.dims.is_playing
+
+
+def test_change_play_axis(ref_view, qtbot):
+    """Make sure changing the play axis stops the current animation"""
```

Comment:
Not that we necessarily want to solve the following in this PR, and I'm glad this solves [this crash](https://github.com/napari/napari/pull/6300#issuecomment-1757696072) (and I think it means we can put a bugfix label on this PR!), but I think we would want to be able to do multiple axes simultaneously in the future. So either we remove/change this test or we can add a note in the docstring, like:

```suggestion
def test_change_play_axis(ref_view, qtbot):
    """Make sure changing the play axis stops the current animation.
    
    Prior to https://github.com/napari/napari/pull/7158, starting a new play
    animation crashed Python. In the future, we may want to allow
    multiple simultaneous play axes [1]_, so this test should be removed
    when we do that.
    
    ..[1] https://github.com/napari/napari/pull/6300#issuecomment-1757696072
    """
```

Reply:
There was no crash. There was a lack of code coverage

Reply:
@Czaki see the linked comment — if you click play on second axis after starting first axis, you get a crash. I just tried it now.

```
WARNING: QThread: Destroyed while thread is still running
16:50:50 : WARNING : MainThread : QThread: Destroyed while thread is still running
```

Reply:
I just confirmed that with this PR, the first play axis stops playing. This is much preferable to the previous behaviour. 😅

Reply:
> @Czaki see the linked comment — if you click play on second axis after starting first axis, you get a crash. I just tried it now.

On my machine it creates warning but napari was still running. And the second click starts the animation again. 
I was my intentional change to require only one click.

Reply:
Ok I have made the message clearer that it only caused crashes in some situations.

```suggestion
def test_change_play_axis(ref_view, qtbot):
    """Make sure changing the play axis stops the current animation.
    
    Prior to https://github.com/napari/napari/pull/7158, starting a new play
    animation resulted in QThread warnings and could crash Python in
    some environments. In the future, we may want to allow multiple
    multiple simultaneous play axes [1]_, so this test should be changed
    or removed when we do that.
    
    ..[1] https://github.com/napari/napari/pull/6300#issuecomment-1757696072
    """
```